From 7d041c71da088b6baf8a9775d3908c56eb4e929f Mon Sep 17 00:00:00 2001
From: "Kothapeta, BikshapathiX" <bikshapathix.kothapeta@intel.com>
Date: Fri, 11 Nov 2022 15:10:13 +0530
Subject: [PATCH] Added valid check for SurfacesPool.

Added valid check for SurfacesPool in MFXGetFreeSurfaceIdx
fixed VTS hal test VtsHalMediaC2V1_0TargetComponentTest

Tracked-On: OAM-104728
Signed-off-by: Kothapeta, BikshapathiX <bikshapathix.kothapeta@intel.com>

diff --git a/c2_utils/src/mfx_defs.cpp b/c2_utils/src/mfx_defs.cpp
index bc0c380..f7cecd5 100755
--- a/c2_utils/src/mfx_defs.cpp
+++ b/c2_utils/src/mfx_defs.cpp
@@ -167,10 +167,11 @@ uint32_t MFXGetSurfaceSize(uint32_t FourCC, uint32_t width, uint32_t height)
 uint32_t MFXGetFreeSurfaceIdx(mfxFrameSurface1 *SurfacesPool, uint32_t nPoolSize)
 {
     MFX_DEBUG_TRACE_FUNC;
-
-    for (uint32_t i = 0; i < nPoolSize; i++) {
-        if (0 == SurfacesPool[i].Data.Locked)
-            return i;
+    if ( SurfacesPool ){
+        for (uint32_t i = 0; i < nPoolSize; i++) {
+            if (0 == SurfacesPool[i].Data.Locked)
+                return i;
+        }
     }
     return MFX_ERR_NOT_FOUND;
 }
-- 
2.38.1

